packages/compose/src/hooks/use-async-list/index.ts

```diff
@@ -55,21 +55,19 @@ function useAsyncList< T >(
 			);
 		}
 		setCurrent( firstItems );
-		let nextIndex = firstItems.length;
 
 		const asyncQueue = createQueue();
-		const append = () => {
+		const append = ( nextIndex: number ) => () => {
 			if ( list.length <= nextIndex ) {
 				return;
 			}
 			setCurrent( ( state ) => [
 				...state,
 				...list.slice( nextIndex, nextIndex + step ),
 			] );
-			nextIndex += step;
-			asyncQueue.add( {}, append );
+			asyncQueue.add( {}, append( nextIndex + step ) );
 		};
-		asyncQueue.add( {}, append );
+		asyncQueue.add( {}, append( firstItems.length ) );
```

Comment:
This would make a great fix for a separate PR.

Reply:
That's true, but there are 8 commits like this in the PR. They are all self-contained fixes that are independent from each other.

At this moment I plan to cleanup the `reinitializeEditor` APIs, something we discussed more than a month ago but never executed. That's the last item that needs to be done before merging.

Reply:
No blockers with this at all, I was just sharing thoughts.